R/bed_window.r

```diff
@@ -60,9 +62,24 @@ bed_window <- function(x, y, genome, ...) {
 
   x <- mutate(x, .start = start, .end = end)
 
-  slop_x <- bed_slop(x, genome, ...)
+  # capture command line args
+  cmd_args <- list(...)
 
-  res <- bed_intersect(slop_x, y, ...)
+  # get arguments for bed_slop and bed_intersect
+  slop_arg_names <- names(formals(bed_slop))
+  intersect_arg_names <- names(formals(bed_intersect))
+
+  # parse supplied args into those for bed_slop or bed_intersect
+  slop_args <- cmd_args[names(cmd_args) %in% slop_arg_names]
+  intersect_args <- cmd_args[names(cmd_args) %in% intersect_arg_names]
+
+  # pass new list of args to bed_slop
+  slop_x <- do.call(bed_slop,
```

Comment:
`purrr::invoke` is a wrapper around `do.call` and could be used, but in this situation it would be identical to using `do.call`. 

```r
# if purrr::invoke was used it would look like this
slop_x <- purrr::invoke(bed_slop,
                  c(list("x" = x, "genome" = genome) , slop_args))

# purrr source code
purrr::invoke
function (.f, .x = NULL, ..., .env = NULL) 
{
    .env <- .env %||% lazyeval::expr_env(.f, parent.frame())
    do.call(.f, c(.x, list(...)), envir = .env)
}